src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs

```diff
@@ -48,13 +48,16 @@ private bool IsValidScanTarget(string file)
 
         public override BinaryAnalyzerContext InitializeGlobalContextFromOptions(AnalyzeOptions options, ref BinaryAnalyzerContext context)
         {
+            base.InitializeGlobalContextFromOptions(options, ref context);
+
             if (this.Telemetry?.TelemetryClient != null)
             {
                 var aggregatingLogger = new AggregatingLogger();
 
                 var ruleTelemetryLogger = new RuleTelemetryLogger(this.Telemetry.TelemetryClient);
                 ruleTelemetryLogger.AnalysisStarted();
 
+                aggregatingLogger.Loggers.Add(context.Logger);
```

Comment:
>aggregatingLogger [](http://example.com/codeflow?start=16&length=17)

Run this test: configure your analysis to specify a SARIF output location. And then debug into this code. What I *think* will happen is that your context object will come back with an aggregating logger that has two things, a console logger and a SARIF logger.

And your change will create this

-> AggregatingLogger
  --> Telemetry Logger
  --> AggregatingLogger
    --> ConsoleLogger
    --> SARIF Logger

Now, I suppose there's no reason *not* to have this nesting, your approach ends up being pretty clean because of it.

Just wanted to make sure we're on the same page with what is happening at runtime and that this is a conscious decision.

Reply:
Yep seems to be more clean than the first attempt. I will investigate the behavior you describing and let's see if I can come up with something elegant.

Reply:
Oh, right. The SARIF log itself is actually initialized in `InitializeOutputs(globalContext)` not in this core global init.

Makes sense! I like the way you approached this, looks perfect.